src/main/java/org/breedinginsight/brapi/v2/BrAPIV2Controller.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * See the NOTICE file distributed with this work for additional information regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.breedinginsight.brapi.v2;
+
+import io.micronaut.http.HttpRequest;
+import io.micronaut.http.HttpResponse;
+import io.micronaut.http.HttpStatus;
+import io.micronaut.http.MediaType;
+import io.micronaut.http.annotation.*;
+import io.micronaut.http.exceptions.HttpStatusException;
+import io.micronaut.security.annotation.Secured;
+import io.micronaut.security.rules.SecurityRule;
+import lombok.extern.slf4j.Slf4j;
+import okhttp3.*;
+import org.brapi.v2.model.core.BrAPIServerInfo;
+import org.brapi.v2.model.core.response.BrAPIServerInfoResponse;
+import org.breedinginsight.api.auth.AuthenticatedUser;
+import org.breedinginsight.api.auth.ProgramSecured;
+import org.breedinginsight.api.auth.ProgramSecuredRoleGroup;
+import org.breedinginsight.api.auth.SecurityService;
+import org.breedinginsight.brapi.v1.controller.BrapiVersion;
+import org.breedinginsight.model.ProgramBrAPIEndpoints;
+import org.breedinginsight.services.ProgramService;
+import org.breedinginsight.services.exceptions.DoesNotExistException;
+
+import javax.inject.Inject;
+import java.io.IOException;
+import java.util.Objects;
+import java.util.UUID;
+
+@Slf4j
+@Controller
+@Secured(SecurityRule.IS_AUTHENTICATED)
+public class BrAPIV2Controller {
+
+    private final SecurityService securityService;
+    private final ProgramService programService;
+
+    @Inject
+    public BrAPIV2Controller(SecurityService securityService, ProgramService programService) {
+        this.securityService = securityService;
+        this.programService = programService;
+    }
+
+
+    @Get(BrapiVersion.BRAPI_V2 + "/serverinfo")
+    @Produces(MediaType.APPLICATION_JSON)
+    @Secured(SecurityRule.IS_ANONYMOUS)
+    public BrAPIServerInfoResponse serverinfo() {
+        BrAPIServerInfo serverInfo = new BrAPIServerInfo();
+        serverInfo.setOrganizationName("Breeding Insight Platform");
+        serverInfo.setServerName("bi-api");
+        serverInfo.setContactEmail("bidevteam@cornell.edu");
+        serverInfo.setOrganizationURL("breedinginsight.org");
+
+        return new BrAPIServerInfoResponse().result(serverInfo);
+    }
+
+    //TODO override program serverinfo endpoint to replace some of the data points (contactEmail, organizationName, organizationURL, etc)
```

Comment:
can remove this comment I think

Reply:
We should talk about how we want that to be handled.  if `/programs/{programId}/brapi/v2/serverinfo` is hit, do we want the response to have the server information of the server we're proxying to or replace the serverinfo metadata with BI info?

Reply:
hmm, not sure. I guess I'm fine leaving it with the server we're proxying unless we had a reason not to.

Reply:
Ok, can always change it later.